tests/model_fields/models.py

```diff
@@ -69,11 +69,18 @@ class WhizIterEmpty(models.Model):
 
 
 class Choiceful(models.Model):
+    class Suit(models.IntegerChoices):
+        DIAMOND = 1, _("Diamond")
+        SPADE = 2, _("Spade")
+        HEART = 3, _("Heart")
+        CLUB = 4, _("Club")
+
     no_choices = models.IntegerField(null=True)
     empty_choices = models.IntegerField(choices=(), null=True)
     with_choices = models.IntegerField(choices=[(1, "A")], null=True)
     empty_choices_bool = models.BooleanField(choices=())
     empty_choices_text = models.TextField(choices=())
+    choices_from_enum = models.IntegerField(choices=Suit)
```

Comment:
In order to keep consistency across Django, I guess it makes sense provide a similar API for the `forms.ChoiceField`.

For instance this test case:
https://github.com/django/django/blob/c914d6cff176ae6bfab2f33a84bcfd45208f1894/tests/forms_tests/field_tests/test_choicefield.py#L93-L102

It would be nice to be able to do `f = ChoiceField(choices=FirstNames)`.

Reply:
attempted to get equivalence there in terms of naming and tests.

Reply:
`Choices` has never been documented or tested with form fields. I'm not sure about introducing it here.

Reply:
@felixxm well above test (`test_choicefield_enumeration`) does test usage of choices on the form field. This merely establishes parity, given that both field types do have choice functionality already.

I do think it makes sense like this, but will remove it if you like.

Reply:
I think that if we're adding this at the model field level, we need to do the same for forms.

Folks will want that. 

More though, `forms.ChoiceField.choices` is [documented as following the model field](https://docs.djangoproject.com/en/4.1/ref/forms/fields/#django.forms.ChoiceField.choices): 

> This argument accepts the same formats as the choices argument to a model field.